msal/src/test/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationJavaTest.java

```diff
@@ -606,24 +605,18 @@ public void onError(@NonNull BaseException exception) {
 
         assertTrue(signInWithPasswordResult.get(10, TimeUnit.SECONDS) instanceof SignInResult.Complete);
 
-        ResultFuture<BaseException> signInExceptionResult = new ResultFuture<>();
-        NativeAuthPublicClientApplication.SignInCallback signInWithPasswordResultCallback2
-                = new NativeAuthPublicClientApplication.SignInCallback() {
-            @Override
-            public void onResult(SignInResult result) {
-
-            }
+        SignInTestCallback signInWithPasswordResultTestCallback = new SignInTestCallback();
 
-            @Override
-            public void onError(@NonNull BaseException exception) {
-                signInExceptionResult.setResult(exception);
-            }
-        };
+        application.signIn(
+                username,
+                password,
+                null,
+                signInWithPasswordResultTestCallback
+        );
+        SignInResult signInError = signInWithPasswordResultTestCallback.get();
```

Comment:
I actually don't love this. I hadn't paid close attention to this before, but an error state should be returned in the `onError()` callback. And the callback signature should change from `exception` to `SignInError`. We can park this for now, but I consider that a future improvement.

Reply:
This improvement might worth a separate work item because it might be lots of changes.
```kotlin
NativeAuthPublicClientApplication.pcaScope.launch {
            try {
                val result = submitPassword(password)
                callback.onResult(result)
            } catch (e: MsalException) {
                Logger.error(TAG, "Exception thrown in submitPassword", e)
                callback.onError(e)
            }
        }
```